bonsai/BudgetTransactions/BudgetTransactions.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  BudgetTransactions.swift
+//  bonsai
+//
+//  Created by hoang on 08.12.2021.
+//
+
+import SwiftUI
+
+struct VisualEffectView: UIViewRepresentable {
+    var effect: UIVisualEffect?
+    func makeUIView(context: UIViewRepresentableContext<Self>) -> UIVisualEffectView { UIVisualEffectView() }
+    func updateUIView(_ uiView: UIVisualEffectView, context: UIViewRepresentableContext<Self>) { uiView.effect = effect }
+}
+
+struct RoundedCorner: Shape {
+
+    var radius: CGFloat = .infinity
+    var corners: UIRectCorner = .allCorners
+
+    func path(in rect: CGRect) -> Path {
+        let path = UIBezierPath(roundedRect: rect, byRoundingCorners: corners, cornerRadii: CGSize(width: radius, height: radius))
+        return Path(path.cgPath)
+    }
+}
+
+extension View {
+    func cornerRadius(_ radius: CGFloat, corners: UIRectCorner) -> some View {
+        clipShape(RoundedCorner(radius: radius, corners: corners))
+    }
+}
+
+
+struct BudgetTransactions: View {
+    
+    private typealias TransactionGroup = [String: [Transaction]]
+    let transactions: [Transaction]
+    private let transactionsByDate: TransactionGroup
+    
+    init(transactions: [Transaction]) {
+        self.transactions = transactions
+        self.transactionsByDate = transactions
+            .reduce(into: TransactionGroup()) {
```

Comment:
.reduce with .filter inside will have complexity of O(n^2) which might work slow on big data
since user might have thousands of transactions, I suggest to rewrite it in O(n)
this might be done with single for loop

Reply:
Changed, but need to reorder dict, need some thoughts how to sort by date, maybe in current loop or after